File: CS50/pset1/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    double changeAmt = get_double("Change owed: ");

    while(changeAmt < 0)
    {
        printf("Change owed: ");
        changeAmt = get_double(0);
    }

    int y=0; //y is the number of 0.25cents coins
    int x=0; //x is the number of 0.10cents coins
    int m=0; //m is the number of 0.05cents coins
    int n=0; //n is the number of 0.01cents coins

    int totalCoins;

    int changeAmtFormatted = (double)changeAmt * 100;

    while(changeAmtFormatted >= 25)
    {
        changeAmtFormatted = changeAmtFormatted - 25;
        y++;
    }

    while(changeAmtFormatted >= 10)
    {
        changeAmtFormatted = changeAmtFormatted - 10;
        x++;
    }

    while(changeAmtFormatted >= 5)
    {
        changeAmtFormatted = changeAmtFormatted - 5;
        m++;
    }

    while(changeAmtFormatted >= 1)
    {
        changeAmtFormatted = changeAmtFormatted - 1;
        n++;
    }

    totalCoins = y + x + m + n;

    printf("%i\n", totalCoins);

}

File: CS50/pset6/cash.py
while True:
    changeOwed = float(input("Change owed: "))
    if changeOwed > 0:
        break

coins = [0,0,0,0]
y, x, m, n = coins

#y is the number of 0.25cents coins
#x is the number of 0.10cents coins
#m is the number of 0.05cents coins
#n is the number of 0.01cents coins

changeAmtFormatted = changeOwed * 100

while changeAmtFormatted >= 25:
    changeAmtFormatted -= 25
    y+=1
while changeAmtFormatted >= 10:
    changeAmtFormatted -= 10
    x+=1
while changeAmtFormatted >= 5:
    changeAmtFormatted -= 5
    m+=1
while changeAmtFormatted >= 1:
    changeAmtFormatted -= 1
    n+=1

coins = y, x, m, n
total_coins = sum(coins)
print(total_coins)